spec/circuitry/subscriber_spec.rb

```diff
@@ -82,62 +82,51 @@
     describe 'when a block is given' do
       let(:block) { ->(_, _) { } }
       let(:logger) { double('Logger', info: nil, warn: nil, error: nil) }
-      let(:mock_sqs) { double('SQS', receive_message: double('Response', body: { 'Message' => messages })) }
+      let(:mock_sqs) { double('SQS', receive_message: response) }
+      let(:response) { double('Aws::SQS::Types::ReceiveMessageResult', messages: messages) }
       let(:messages) { [] }
 
       before do
         allow(Circuitry.config).to receive(:logger).and_return(logger)
         allow(subject).to receive(:sqs).and_return(mock_sqs)
-        allow(subject).to receive(:loop) do |&block|
-          block.call
+
+        call_count = 0
+        allow(subject).to receive(:subscribed?) do
+          call_count += 1
+          call_count == 1
         end
       end
 
       describe 'when AWS credentials are set' do
         before do
-          allow(Circuitry.config).to receive(:aws_options).and_return(aws_access_key_id: 'key', aws_secret_access_key: 'secret', region: 'region')
+          allow(Circuitry.config).to receive(:aws_options).and_return(access_key_id: 'key', secret_access_key: 'secret', region: 'region')
         end
 
         it 'subscribes to SQS' do
           subject.subscribe(&block)
-          expect(mock_sqs).to have_received(:receive_message).with(queue, any_args)
-        end
-
-        describe 'when a connection error is raised' do
-          before do
-            allow(subject).to receive(:receive_messages).and_raise(described_class::CONNECTION_ERRORS.first, 'Forbidden')
-          end
-
-          it 'raises a wrapped error' do
-            expect { subject.subscribe(&block) }.to raise_error(Circuitry::SubscribeError)
-          end
-
-          it 'logs an error' do
-            subject.subscribe(&block) rescue nil
-            expect(logger).to have_received(:error)
-          end
+          expect(mock_sqs).to have_received(:receive_message).with(hash_including(queue_url: queue))
         end
 
-        describe 'when a temporary error is raised' do
-          before do
-            allow(mock_sqs).to receive(:receive_message).and_raise(described_class::TEMPORARY_ERRORS.first, 'Server Error')
-          end
-
-          it 'does not raise an error' do
-            expect { subject.subscribe(&block) }.to_not raise_error
-          end
-
-          it 'logs info' do
-            subject.subscribe(&block) rescue nil
-            expect(logger).to have_received(:info)
-          end
-        end
+        # describe 'when a connection error is raised' do
```

Comment:
This pertains to the messaging in my PR body.

> This is a WIP because I need to check into the CONNECTION_ERRORS code to see if I need to map that over from the fog-aws gem, or if the aws-sdk gem takes care of that logic for us already. I think it may be the latter.